integrations/vite/index.test.ts

```diff
@@ -14,680 +14,686 @@ import {
+        expect(styles).toContain(candidate`font-bold`)
+      })
+    },
+  )
+
+  test(
+    'watch mode',
+    {
+      fs: {
+        'package.json': json`{}`,
+        'pnpm-workspace.yaml': yaml`
+          #
+          packages:
+            - project-a
+        `,
+        'project-a/package.json': txt`
+          {
+            "type": "module",
+            "dependencies": {
+              "@tailwindcss/vite": "workspace:^",
+              "tailwindcss": "workspace:^"
+            },
+            "devDependencies": {
+              ${transformer === 'lightningcss' ? `"lightningcss": "^1.26.0",` : ''}
+              "vite": "^6"
             }
-          `,
-          'project-b/src/index.html': html`
-            <div class="flex" />
-          `,
-          'project-b/src/index.js': js`
-            const className = "content-['project-b/src/index.js']"
-            module.exports = { className }
-          `,
-        },
+          }
+        `,
+        'project-a/vite.config.ts': ts`
+          import tailwindcss from '@tailwindcss/vite'
+          import { defineConfig } from 'vite'
+
+          export default defineConfig({
+            build: { cssMinify: false },
+            plugins: [tailwindcss()],
+          })
+        `,
+        'project-a/index.html': html`
+          <head>
+            <link rel="stylesheet" href="./src/index.css" />
+          </head>
+          <body>
+            <div class="underline text-primary">Hello, world!</div>
+          </body>
+        `,
+        'project-a/tailwind.config.js': js`
+          export default {
+            content: ['../project-b/src/**/*.js'],
+          }
+        `,
+        'project-a/src/index.css': css`
+          @import 'tailwindcss/theme' theme(reference);
+          @import 'tailwindcss/utilities';
+          @import './custom-theme.css';
+          @config '../tailwind.config.js';
+          @source '../../project-b/src/**/*.html';
+        `,
+        'project-a/src/custom-theme.css': css`
+          /* Will be overwritten later */
+          @theme {
+            --color-primary: black;
+          }
+        `,
+        'project-b/src/index.html': html`
+          <div class="flex" />
+        `,
+        'project-b/src/index.js': js`
+          const className = "content-['project-b/src/index.js']"
+          module.exports = { className }
+        `,
       },
-      async ({ root, spawn, fs }) => {
-        await spawn(`pnpm vite build --watch`, {
-          cwd: path.join(root, 'project-a'),
-        })
-
-        let filename = ''
-        await retryAssertion(async () => {
-          let files = await fs.glob('project-a/dist/**/*.css')
-          expect(files).toHaveLength(1)
-          filename = files[0][0]
-        })
+    },
+    async ({ root, spawn, fs, expect }) => {
+      let process = await spawn('pnpm vite build --watch', {
+        cwd: path.join(root, 'project-a'),
+      })
+      await process.onStdout((m) => m.includes('built in'))
```

Comment:
This is an example where we weren't properly waiting for the initial build to be ready.